controllers/glance_controller.go

```diff
@@ -822,7 +823,39 @@ func (r *GlanceReconciler) ensureRegisteredLimits(
 			ResourceName: lName,
 			DefaultLimit: lValue,
 		}
-		_, err = o.CreateRegisteredLimit(r.Log, m)
+		_, err = o.CreateOrUpdateRegisteredLimit(r.Log, m)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// registeredLimitsDelete - cleanup registered limits in keystone
+func (r *GlanceReconciler) registeredLimitsDelete(
+	ctx context.Context,
+	h *helper.Helper,
+	instance *glancev1.Glance,
+	quota map[string]int,
+) error {
+
+	// get admin
+	var err error
+	keystoneAPI, err := keystonev1.GetKeystoneAPI(ctx, h, instance.Namespace, map[string]string{})
+
+	if err != nil {
+		return err
+	}
+	o, _, err := glancev1.GetAdminServiceClient(ctx, h, keystoneAPI)
+	if err != nil {
+		return err
+	}
+	fetchRegLimits, err := o.ListRegisteredLimitsByServiceID(r.Log, instance.Status.ServiceID)
```

Comment:
@konan-abhi I'm able to fetch the limits as you suggested, get the ID (see L858) and call the delete.